packages/out-of-tree-platforms/index.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ * @format
+ */
+
+import type {CustomResolver} from 'metro-resolver';
+
+type ResolverConfig = {
+  platformNameMap: {[platform: string]: string},
+};
+
+const getPlatformResolver = (config: ResolverConfig): CustomResolver => {
```

Comment:
What do you think about adding one sentence of explanation from where we get `platformExtension`?